LibOS/shim/src/shim_init.c

```diff
@@ -492,14 +496,15 @@ int init_manifest (PAL_HANDLE manifest_handle)
         sys_printf("Unable to read manifest file: %s\n", errstring);
         goto fail;
     }
-    root_config = new_root_config;
 
+    root_config = new_root_config;
     return 0;
+
 fail:
-    if (bkeep_unmap_on_fail)
-        bkeep_munmap(addr, ALIGN_UP(size), VMA_INTERNAL);
-    if (stream_unmap_on_fail)
-        DkStreamUnmap(addr, size);
+    if (addr == PAL_CB(manifest_preload.start))
+        DkStreamUnmap(addr, map_size);
+    if (bkeep_munmap(addr, map_size, MAP_FLAGS) < 0)
+        bug();
     free(new_root_config);
```

Comment:
`new_root_config` may be uninitialized here. The previous version of this code didn't contain this bug, please revert changes to this variable.